app/code/Magento/InventoryConfiguration/Model/StockItemConfiguration.php

```diff
@@ -324,10 +308,9 @@ public function isManageStock(): bool
     /**
      * @inheritdoc
      */
-    public function setManageStock(bool $manageStock): StockItemConfigurationInterface
+    public function setManageStock(bool $manageStock): void
     {
         $this->stockItem = $this->stockItem->setManageStock($manageStock);
```

Comment:
I don't think that we need to re-assign `$this->stockItem`
As it's updated after the set operations.
@bartekszymanski  could you please fix such assignment in all these setter methods and leave just
`$this->stockItem->setManageStock($manageStock);`